depmapomics/ccle.py

```diff
@@ -0,0 +1,1423 @@
+
+    tc = TaigaClient()
+
+    if prevgenecn == "ccle":
+        prevgenecn = tc.get(name=TAIGA_ETERNAL, file="CCLE_gene_cn")
+
+    tracker = pd.DataFrame()
+    if trackerobj is not None:
+        tracker = trackerobj.read_seq_table()
+
+    assert len(tracker) != 0, "broken source for sample tracker"
+
+    # doing wes
+    folder = os.path.join("output", samplesetname, "wes_")
+    if redoWES:
+        print("doing wes")
+        priority = tracker[
+            (tracker.datatype == "wes") & (tracker.prioritized == 1)
+        ].index.tolist()
+        todropwes = (
+            todrop
+            + tracker[
+                (tracker.datatype == "wes") & (tracker.blacklist == 1)
+            ].index.tolist()
+        )
+        (
+            wessegments,
+            genecn,
+            wesfailed,
+            wes_purecn_segments,
+            wes_purecn_genecn,
+            wes_loh,
+            wes_purecn_table,
+            _,
+        ) = cn.postProcess(
+            wesrefworkspace,
+            setEntity=wessetentity,
+            sampleset=AllSamplesetName if AllSamplesetName else samplesetname,
+            todrop=todropwes,
+            save_output=folder,
+            priority=priority,
+            genechangethresh=genechangethresh,
+            segmentsthresh=segmentsthresh,
+            maxYchrom=maxYchrom,
+            **kwargs,
+        )
+
+        # annotating source
+        for v in set(wessegments[SAMPLEID]):
+            wessegments.loc[
+                wessegments[wessegments[SAMPLEID] == v].index, "Source"
+            ] = tracker.loc[v, "source"]
+
+        wessegments.Source = wessegments.Source.replace(source_rename)
+        wessegments.Source += " WES"
+        # wes_purecn_segments.to_csv(folder + "purecn_segments_latest.csv", index=False)
+        # wes_purecn_genecn.to_csv(folder + "purecn_genecn_latest.csv")
+        # wes_loh.to_csv(folder + "purecn_loh_latest.csv")
+        # wes_purecn_table.to_csv(folder + "purecn_table_latest.csv")
+        # subset and rename to PR-indexed matrices
+        # assuming no wgs
+        pr_table = trackerobj.read_pr_table()
+        renaming_dict = dict(list(zip(pr_table.CDSID, pr_table.index)))
+        wessegments_pr = (
+            wessegments[wessegments[SAMPLEID].isin(set(renaming_dict.keys()))]
+            .replace({SAMPLEID: renaming_dict})
+            .reset_index(drop=True)
+        )
+        # wessegments_pr.to_csv(folder + "segments_all_profile.csv", index=False)
+        # wescn_pr.to_csv(folder + "genecn_all_profile.csv")
+    else:
+        print("bypassing WES using folder: " + (wesfolder if wesfolder else folder))
+        wesfailed = h.fileToList((wesfolder if wesfolder else folder) + "failed.txt")
+        wessegments = pd.read_csv(folder + "segments_all.csv")
+        genecn = pd.read_csv(folder + "genecn_all.csv", index_col=0)
+        wessegments_pr = (
+            wessegments[wessegments[SAMPLEID].isin(set(renaming_dict.keys()))]
```

Comment:
how do you access the renaming dict?

Reply:
it's defined at the top of the function now